test/OpenTelemetry.Exporter.Geneva.Tests/GenevaLogExporterAFDCorrelationTests.cs

```diff
@@ -0,0 +1,392 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable disable
+
+using System.Net.Sockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Context;
+using OpenTelemetry.Exporter.Geneva.MsgPack;
+using OpenTelemetry.Logs;
+using Xunit;
+
+namespace OpenTelemetry.Exporter.Geneva.Tests;
+
+[Collection(nameof(GenevaCorrelationFixture))]
+public class GenevaLogExporterAFDCorrelationTests
+{
+    [Fact(Skip = "Run locally to evaluate with multi-threaded scenario.")]
+    public void AFDCorrelationIdLogProcessor_MultithreadedAccess_HandlesGracefully()
+    {
+        var path = string.Empty;
+        Socket senderSocket = null;
+        Socket receiverSocket = null;
+
+        try
+        {
+            var exporterOptions = new GenevaExporterOptions();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                exporterOptions.ConnectionString = "EtwSession=OpenTelemetry;PrivatePreviewEnableAFDCorrelationIdEnrichment=true";
+            }
+            else
+            {
+                path = GenerateTempFilePath();
+                exporterOptions.ConnectionString = "Endpoint=unix:" + path + ";PrivatePreviewEnableAFDCorrelationIdEnrichment=true";
+                var endpoint = new UnixDomainSocketEndPoint(path);
+                senderSocket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.IP);
+                senderSocket.Bind(endpoint);
+                senderSocket.Listen(1);
+            }
```

Comment:
nit: it looks like this block is repeated in multiple tests. consider moving this to a static helper method.

Reply:
We could follow up on this, as it's part of the other exporter test too. It needs refactoring.